src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -1369,13 +1409,49 @@ class ArrayGenerator : public Generator<bsoncxx::array::value> {
           _node{node},
           _generatorArgs{generatorArgs},
           _valueGen{valueGenerator<false, UniqueAppendable>(node["of"], generatorArgs, parsers)},
-          _nTimesGen{intGenerator(extract(node, "number", "^Array"), generatorArgs)} {}
+          _nTimesGen{intGenerator(extract(node, "number", "^Array"), generatorArgs)},
+          _distinct {node["distinct"] ? node["distinct"].maybe<bool>().value_or(false) : false} {}
 
     bsoncxx::array::value evaluate() override {
         bsoncxx::builder::basic::array builder{};
         auto times = _nTimesGen->evaluate();
-        for (int i = 0; i < times; ++i) {
-            _valueGen->append(builder);
+        if (_distinct) {
+            // Choose initial bucket size of 2N to minimize hash collisions and resizes.
```

Comment:
Still somewhat confusing with the discrepancy between 'N' and 'times'... I actually don't really like 'times' since it kinda implies 'number of times to generate', which is not equivalent to 'number of output results'. Maybe replace both 'times' and 'N' with 'outputSize' or 'nOutput'?

Reply:
I'd note that nTimes was existing, but also internal, so I'm all in favor of clearer names. The input is `number`, so Charlie's suggestions make sense to me.